app/scripts/views/settings/communication_preferences.js

```diff
@@ -39,22 +39,35 @@ function (Cocktail, Xss, Constants, MarketingEmailErrors, BaseView, FormView,
     beforeRender: function () {
       var self = this;
       var emailPrefs = self.getMarketingEmailPrefs();
-
       return emailPrefs.fetch()
         .fail(function (err) {
           if (MarketingEmailErrors.is(err, 'UNKNOWN_EMAIL')) {
             // user has not yet opted in to Basket yet. Ignore.
             return;
           }
+          if (MarketingEmailErrors.is(err, 'SERVICE_UNAVAILABLE')) {
+            // the service is not available, ask to retry later
+            $('.spinner').hide();
+            $('.error').html(err).show();
+            setTimeout(function() {
+              self.navigate('/');
```

Comment:
Let's also log any error that occurs, it doesn't look like that is being done unless the user is [redirected to /unexpected_error`](https://github.com/mozilla/fxa-content-server/blob/f50336ed44237d2b1747f8eef73e5f5ea36c68e8/app/scripts/router.js#L285), which is both unfortunate and unexpected.